dask/array/random.py

```diff
@@ -165,7 +162,6 @@ def _broadcast_any(ar, shape, chunks):
                     arg.append(ar)
                 else:
                     if isinstance(ar, Array):
-                        dependencies.append(ar)
```

Comment:
Adding the full `ar` to `dependencies` (in a loop over every block) is unnecessary. We've already added the broadcast+rechunked array to dependencies [here](https://github.com/dask/dask/pull/7487/files#diff-bacbead89e1b64dbdf54cbe8878e9a52445945d2c2710b2a11edc65364c6b16dR122). This non-broadcast version isn't directly referenced (though it's indirectly referenced by the broadcast+rechunked version), so it's just bloat in the HLG.